pkg/bindings/connection.go

```diff
@@ -85,26 +135,40 @@ func pingNewConnection(ctx context.Context) error {
 	return errors.Errorf("ping response was %q", response.StatusCode)
 }
 
-// newConnection takes a scheme and address and creates a connection from it
-func newConnection(scheme, address string) Connection {
-	client := http.Client{
+func sshClient(_url *url.URL, identity string) (*http.Client, error) {
+	auth, err := publicKey(identity)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to parse identity %s: %v\n", identity)
+	}
+	bastion, err := ssh.Dial("tcp",
```

Comment:
This doesn't support `~/.ssh/config` and yet `InsecureIgnoreHostKey()` is insecure.
I suggest porting over `docker system dial-stdio`.

Reply:
@AkihiroSuda We trying to actively avoid running ssh on the client.  We did that in the previous version of the API and it was very painful for users. I can work on not using `InsecureIgnoreHostKey()`, or at least make it optional.

Reply:
Without using ssh binary, isn't it hard to support bastions?

Reply:
sshd is only required on the bastion, no ssh binary on the client.  I updated the code to allow for known_hosts and secure connections.

Reply:
Sorry, I meant `ProxyCommand` stuff